classes/WP_CLI/Action/Delete.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * WP-CLI command: action-scheduler action delete
+ */
+class ActionScheduler_WPCLI_Action_Delete_Command extends ActionScheduler_WPCLI_Command {
+
+	/** @var int[] */
+	protected $action_ids = array();
+
+	/** @var array<string, int> */
+	protected $action_counts = array(
+		'deleted' => 0,
+		'total'   => 0,
+	);
+
+	/**
+	 * Construct.
+	 *
+	 * @param string[]              $args       Positional arguments.
+	 * @param array<string, string> $assoc_args Keyed arguments.
+	 */
+	public function __construct( array $args, array $assoc_args ) {
+		parent::__construct( $args, $assoc_args );
+
+		$this->action_ids             = array_map( 'absint', $args );
+		$this->action_counts['total'] = count( $this->action_ids );
+
+		add_action( 'action_scheduler_deleted_action', array( $this, 'action__deleted' ) );
```

Comment:
we should have a check on whether `WP_CLI` is set before hooking this action.

Reply:
Because of the way the `action` subcommands are being loaded, they only load in a `WP_CLI` environment, but I will add a check to the abstract class for redundancy.